GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3246,13 +3253,22 @@ private void HandleDrop(DragEventArgs e)
                 return;
             }
 
-            foreach (string path in paths)
+            foreach (string path in paths.Where(IsFileExistingInRepo))
             {
+                if (CommitInfoTabControl.SelectedTab != TreeTabPage)
+                {
+                    CommitInfoTabControl.SelectedTab = TreeTabPage;
+                }
+
                 if (fileTree.SelectFileOrFolder(path))
                 {
                     return;
                 }
```

Comment:
I don't think that will work because `fileTree.SelectFileOrFolder(path)` doesn't select the file if the file tree is not loaded so you have to select the tab before.

Reply:
It did work when I tried it, but the comment above indicates that there are other situations too